### ISSUE TITLE
Status code always 200 // Response headers always empty

### ISSUE THREAD

**Issue by ppfeister** [FlareSolverr/FlareSolverr#1162]
### Have you checked our README?

- [X] I have checked the README

### Have you followed our Troubleshooting?

- [X] I have followed your Troubleshooting

### Is there already an issue for your problem?

- [X] I have checked older issues, open and closed

### Have you checked the discussions?

- [X] I have read the Discussions

### Environment

```markdown
- FlareSolverr version:3.3.17
- Last working FlareSolverr version:Unk
- Operating system:All
[ other data not applicable ]
```

### Description

Status codes are not directly supported by Selenium and therefore FlareSolverr always returns 200.
Response headers are not directly supported by Selenium and therefore FlareSolverr always returns empty headers.

From [flaresolverr_service.py](https://github.com/FlareSolverr/FlareSolverr/blob/master/src/flaresolverr_service.py):
```python
challenge_res.status = 200  # todo: fix, selenium not provides this info
challenge_res.headers = {}  # todo: fix, selenium not provides this info
```

I don't believe this is documented anywhere outside of the code, so new users may encounter issues while debugging. This should serve as an easily searchable Issue to save their sanity until this is resolved.

### Logged Error Messages

```text
[ log not applicable ]
```

**Comment by ppfeister** [comment]
Hopefully I can figure out a workaround with cdp and the webdriver logs before I lose _my own_ sanity

**Comment by asulwer** [comment]
this issue has been discussed and they are not including support for this.

[Selenium Devs say no](https://github.com/seleniumhq/selenium-google-code-issue-archive/issues/141)

although, i am pursuing this discussion

[??possible solution??](https://stackoverflow.com/questions/6509628/how-to-get-http-response-code-using-selenium-webdriver)

**Comment by ppfeister** [comment]
Hey @asulwer !

We're aware that it's a selenium issue

My personal goal was (& is when free) to shoehorn support into flaresolverr by interacting with the browser behind selenium directly, rather than interacting with selenium (since they don't want to support it, as linked)

While Selenium doesn't support those items directly, the devtools for whichever browser are still somewhat exposed, giving us a _theoretical_ side channel, even if it's not pretty

It's a low priority right now but something I'd like to poke at eventually

**Comment by asulwer** [comment]
another post states that it isn't possible with chromedriver but with firefox it is?  i 'briefly' made an attempt at this with no success.

https://stackoverflow.com/questions/6509628/how-to-get-http-response-code-using-selenium-webdriver
https://stackoverflow.com/questions/5799228/how-to-get-status-code-by-using-selenium-py-python-code

**Comment by GravuTrad** [comment]
hello.
Is it why i have each minute this kind of troublesome "ping"?:
example:
2024-06-26 21:34:01 INFO     141.98.11.179 GET http://180.164.250.20:8191/ 200 OK
2024-06-26 21:35:01 INFO     141.98.11.179 GET http://180.164.250.20:8191/ 200 OK
2024-06-26 21:36:01 INFO     141.98.11.179 GET http://180.164.250.20:8191/ 200 OK

**Comment by ilike2burnthing** [comment]
No, that's presumably because you have FlareSolverr exposed the internet and someone (more likely a bot) is pinging to see if they can access it. Don't expose FlareSolverr to the internet.

**Comment by GravuTrad** [comment]
so i can't use one of my cloud servers to host it and use it anywhere? possible to close it to only one ip? (i don't see any environment variable for this kind of use)

**Comment by ilike2burnthing** [comment]
No, that wouldn't work anyway, the cookies would be invalid.

**Comment by GravuTrad** [comment]
an on a vm hosted on my box, with the lan of the box?

**Comment by ilike2burnthing** [comment]
See the troubleshooting wiki for advice on how to use FlareSolverr. Different IPs, different hosts, etc. can result in invalid cookies. Opening ports, whether for a VM or physical hardware, is a security risk.

**Comment by cubetech126** [comment]
It would be nice to include custom headers support.

Example:

```json
{
  "cmd": "request.get",
  "url": "https://www.google.com",
  "maxTimeout": 60000,
  "headers": {
      "X-custom": "some_value"
  }
}
```

**Comment by deadmeat333** [comment]
HI guys, any update on this?

**Comment by Theta-Dev** [comment]
It is possible to get the status code from the current page by running this JS code:

```py
result = driver.execute_script(
            "const navigationData = window.performance.getEntriesByType('navigation')[0]; return navigationData.responseStatus;"
)
status_code = int(result)
```

**Comment by cubetech126** [comment]
Still no workaround to make requests with custom headers?

**Comment by trucchuspear-policymogul** [comment]
Just checking if this is the same issue?

Using 3.5.0 Docker image, I post the following:

```
{
    "cmd": "request.get",
    "url": "https://lordsbusiness.parliament.uk/ItemOfBusiness?itemOfBusinessId=159619&sectionId=38&businessPaperDate=2025-09-03",
    "maxTimeout": 60000'
}
```

In the response, I see

```
{
    "status": "ok",
    "message": "Challenge not detected!",
    "solution": {
        "url": "https://lordsbusiness.parliament.uk/ItemOfBusiness?itemOfBusinessId=159619&sectionId=38&businessPaperDate=2025-09-03",
        "status": 200,
       ...
     },
    ...
)
```

but when I visit https://lordsbusiness.parliament.uk/ItemOfBusiness?itemOfBusinessId=159619&sectionId=38&businessPaperDate=2025-09-03 locally on my web browser, the true HTTP status is 500
